BSP/STM32F413/Src/BSP_GPIO.c

```diff
@@ -37,6 +37,31 @@ void BSP_GPIO_Init(port_t port, uint16_t mask, direction_t direction, bool pull_
     GPIO_Init(portHandle, &GPIO_InitStruct);
 }
 
+/**
+ * @brief   Initializes a GPIO port
+ * @param   port - port to initialize
+ * @param	mask - pins
+ * @param	direction - input or output 
+ * @param	pull_up - pulled up or not
+ * @return  None
+ */ 
+void BSP_GPIO_Init_PullUp(port_t port, uint16_t mask, direction_t direction, bool pull_up){

```

Comment:
u could get rid of this and find and replace existing GPIO inits to support pullup